samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/UpdateAssessmentQuestionsOrder.java

```diff
@@ -0,0 +1,167 @@
+package org.sakaiproject.tool.assessment.ui.listener.author;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.faces.event.AbortProcessingException;
+import javax.faces.event.ActionEvent;
+import javax.faces.event.ActionListener;
+
+import org.sakaiproject.tool.assessment.data.dao.assessment.ItemData;
+import org.sakaiproject.tool.assessment.data.ifc.assessment.ItemDataIfc;
+import org.sakaiproject.tool.assessment.facade.AgentFacade;
+import org.sakaiproject.tool.assessment.facade.ItemFacade;
+import org.sakaiproject.tool.assessment.facade.PublishedSectionFacade;
+import org.sakaiproject.tool.assessment.facade.SectionFacade;
+import org.sakaiproject.tool.assessment.services.ItemService;
+import org.sakaiproject.tool.assessment.services.assessment.AssessmentService;
+import org.sakaiproject.tool.assessment.services.assessment.PublishedAssessmentService;
+import org.sakaiproject.tool.assessment.ui.bean.author.AssessmentBean;
+import org.sakaiproject.tool.assessment.ui.bean.author.AuthorBean;
+import org.sakaiproject.tool.assessment.ui.bean.delivery.ItemContentsBean;
+import org.sakaiproject.tool.assessment.ui.bean.delivery.SectionContentsBean;
+import org.sakaiproject.tool.assessment.ui.listener.util.ContextUtil;
+
+public class UpdateAssessmentQuestionsOrder implements ActionListener {
+
+	ItemService delegate;
+	AssessmentService assessdelegate;
+	PublishedAssessmentService publisheddelegate;
+	
+	public void processAction(ActionEvent ae) throws AbortProcessingException {
+		
+		AuthorBean author = (AuthorBean) ContextUtil.lookupBean("author");
+		this.delegate = new ItemService();
+		
+		boolean isEditPendingAssessmentFlow = author
+				.getIsEditPendingAssessmentFlow();
+
+		AssessmentBean assessmentBean = (AssessmentBean) ContextUtil.lookupBean("assessmentBean");
+		
+		
+		boolean published = isEditPendingAssessmentFlow ? false : true;
+		
+		if(published)
+			publisheddelegate = new PublishedAssessmentService();
+		else assessdelegate = new AssessmentService();
```

Comment:
This is a little nit-picky, but it's better to have braces on single line conditionals. Omitting them can lead to problems down the road.